[firebase_messaging-15.2.0]:  deprecated Warinings and errors

### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Which plugins are affected?

Messaging

### Which platforms are affected?

Android

### Description

/.pub-cache/hosted/pub.dev/firebase_messaging-15.2.0/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingUtils.java:50: warning: [deprecation] getTo() in RemoteMessage has been deprecated
    if (remoteMessage.getTo() != null) {
                     ^
/.pub-cache/hosted/pub.dev/firebase_messaging-15.2.0/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingUtils.java:51: warning: [deprecation] getTo() in RemoteMessage has been deprecated
      messageMap.put(KEY_TO, remoteMessage.getTo());
                                          ^
2 warnings

### Reproducing the issue

/.pub-cache/hosted/pub.dev/firebase_messaging-15.2.0/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingUtils.java:50: warning: [deprecation] getTo() in RemoteMessage has been deprecated
    if (remoteMessage.getTo() != null) {
                     ^
/.pub-cache/hosted/pub.dev/firebase_messaging-15.2.0/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingUtils.java:51: warning: [deprecation] getTo() in RemoteMessage has been deprecated
      messageMap.put(KEY_TO, remoteMessage.getTo());
                                          ^
2 warnings

### Firebase Core version

3.10.0

### Flutter Version

3.27.2

### Relevant Log Output

```shell
/.pub-cache/hosted/pub.dev/firebase_messaging-15.2.0/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingUtils.java:50: warning: [deprecation] getTo() in RemoteMessage has been deprecated
    if (remoteMessage.getTo() != null) {
                     ^
/.pub-cache/hosted/pub.dev/firebase_messaging-15.2.0/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingUtils.java:51: warning: [deprecation] getTo() in RemoteMessage has been deprecated
      messageMap.put(KEY_TO, remoteMessage.getTo());
                                          ^
2 warnings
```

### Flutter dependencies

<!--- Look below for instructions on how to share your Flutter Dependencies. --->

  firebase_app_installations: ^0.3.2
  firebase_core: ^3.10.0
  firebase_crashlytics: ^4.3.0
  firebase_messaging: ^15.2.0
  firebase_remote_config: ^5.3.0<br>


### Additional context and comments

_No response_

--- a/packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingUtils.java
+++ b/packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingUtils.java
@@ -35,6 +35,9 @@ class FlutterFirebaseMessagingUtils {
   private static final String KEY_TO = "to";
   private static final String KEY_TTL = "ttl";
 
+  // We are using a deprecated method 'getTo' which is not being replaced by any other method.
+  // Keeping this method for backward compatibility.
+  @SuppressWarnings("deprecation")
   static Map<String, Object> remoteMessageToMap(RemoteMessage remoteMessage) {
     Map<String, Object> messageMap = new HashMap<>();
     Map<String, Object> dataMap = new HashMap<>();
@@ -59,7 +62,7 @@ class FlutterFirebaseMessagingUtils {
       messageMap.put(KEY_MESSAGE_TYPE, remoteMessage.getMessageType());
     }
 
-    if (remoteMessage.getData().size() > 0) {
+    if (!remoteMessage.getData().isEmpty()) {
       Set<Map.Entry<String, String>> entries = remoteMessage.getData().entrySet();
       for (Map.Entry<String, String> entry : entries) {
         dataMap.put(entry.getKey(), entry.getValue());

